src/CodeGen/ODataT4CodeGenerator.tt

```diff
@@ -27,7 +27,9 @@ public static class Configuration
 
 	// This flag indicates whether to ignore unexpected elements and attributes in the metadata document and generate
 	// the client code if any. The value must be set to true or false.
-	public const bool IgnoreUnexpectedElementsAndAttributes = true;
+	public const bool IgnoreUnexpectedElementsAndAttributes = true;	
+	// Here you can set custom http headers like this: [{'headername': 'headervalue'}, {'headername': 'headervalue' }]
+	public const string CustomHttpHeaders = "[]";

```

Comment:
Might be easier to simplify this to remove the square brackets and single quotes. Note that we need to support escaped double-quotes within the string for key-values containing double quotes.